File: app/src/main/java/com/mdgd/pokemon/ui/pokemon/adapter/PokemonPropertyViewHolder.java
package com.mdgd.pokemon.ui.pokemon.adapter;

import android.view.View;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.mdgd.pokemon.ui.pokemon.items.PokemonProperty;

public abstract class PokemonPropertyViewHolder<T extends PokemonProperty> extends RecyclerView.ViewHolder {

    public PokemonPropertyViewHolder(@NonNull View itemView) {
        super(itemView);
    }


    public void setupSubscriptions() {
    }

    public void clearSubscriptions() {
    }

    public abstract void bind(T property, int position);
}

File: app/src/main/java/com/mdgd/pokemon/ui/pokemon/adapter/PokemonPropertyViewHolder.kt
package com.mdgd.pokemon.ui.pokemon.adapter

import android.view.View
import androidx.recyclerview.widget.RecyclerView
import com.mdgd.pokemon.ui.pokemon.infra.PokemonProperty

abstract class PokemonPropertyViewHolder<T : PokemonProperty>(itemView: View) : RecyclerView.ViewHolder(itemView) {
    fun setupSubscriptions() {}

    fun clearSubscriptions() {}

    abstract fun bind(property: T, position: Int)
}
